src/overlays/gamestates/ovl_play/m_play.c

```diff
@@ -0,0 +1,868 @@
+    gSegments[2] = (uintptr_t)OS_K0_TO_PHYSICAL(play->unk_010C);
+    play->state.unk_9C = 2;
+
+    if (zurumode_flag >= 2) {
+        var_v1 = Pause_proc(&play->pause, &play->state.input[1]) == 0;
+    } else {
+        var_v1 = 0;
+    }
+
+    play->state.unk_9D = 0x8E;
+    play->state.unk_9C = 0;
+
+    if (var_v1 == 0) {
+        play->state.unk_9C = 1;
+        Game_play_fbdemo_wipe_proc(play);
+        play->state.unk_9C = 2;
+        Game_play_fbdemo_proc(play);
+        play->state.unk_9C = 3;
+        if (play->unk_1EE2 != 3) {
+            Game_play_move_fbdemo_not_move(play);
+        }
+    }
+
+    if (play->unk_1CBC.unk_0C == 0) {
+        play->state.unk_9D = 0x92;
+        play->state.unk_9C = 1;
+        Game_play_camera_proc(play);
+        play->state.unk_9C = 2;
+        mPO_business_proc(play);
+        play->state.unk_9C = 3;
+        mTRC_move(play);
+        play->state.unk_9C = 4;
+        Balloon_move(play);
+    }
+
+    play->state.unk_9D = 0x93;
+    play->state.unk_9C = 1;
+    Global_kankyo_set(play, &play->kankyo, &play->lightCtx);
+    play->state.unk_9C = 2;
+    mEnv_WindMove();
+    play->state.unk_9C = 3;
+    watch_my_step_move(play);
+    play->state.unk_9C = 4;
+    banti_move(play);
+    play->state.unk_9C = 5;
+    title_demo_move(play);
+    play->state.unk_9D = 0x95;
+    play->state.unk_9C = 0;
+}
+
+void func_80803810_jp(PlayState* play, GraphicsContext* gfxCtx) {
+    void* temp_v0 = play->unk_0110[0].segment;
+
+    gSegments[4] = (uintptr_t)OS_K0_TO_PHYSICAL(temp_v0);
+    gSegments[2] = (uintptr_t)OS_K0_TO_PHYSICAL(play->unk_010C);
+
+    OPEN_DISPS(gfxCtx);
+
+    gSPSegment(POLY_OPA_DISP++, 0x00, NULL);
+    gSPSegment(POLY_XLU_DISP++, 0x00, NULL);
+    gSPSegment(OVERLAY_DISP++, 0x00, NULL);
+    gSPSegment(UNK_2B0_DISP++, 0x00, NULL);
+    gSPSegment(UNK_2C0_DISP++, 0x00, NULL);
+    gSPSegment(LIGHT_DISP++, 0x00, NULL);
+
+    gSPSegment(POLY_OPA_DISP++, 0x04, temp_v0);
+    gSPSegment(POLY_XLU_DISP++, 0x04, temp_v0);
+    gSPSegment(OVERLAY_DISP++, 0x04, temp_v0);
+    gSPSegment(UNK_2B0_DISP++, 0x04, temp_v0);
+    gSPSegment(UNK_2C0_DISP++, 0x04, temp_v0);
+    gSPSegment(LIGHT_DISP++, 0x04, temp_v0);
+
+    gSPSegment(POLY_OPA_DISP++, 0x02, play->unk_010C);
+    gSPSegment(POLY_XLU_DISP++, 0x02, play->unk_010C);
+    gSPSegment(OVERLAY_DISP++, 0x02, play->unk_010C);
+    gSPSegment(UNK_2B0_DISP++, 0x02, play->unk_010C);
+    gSPSegment(UNK_2C0_DISP++, 0x02, play->unk_010C);
+    gSPSegment(LIGHT_DISP++, 0x02, play->unk_010C);
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void setupFog(PlayState* play, GraphicsContext* gfxCtx) {
+    OPEN_DISPS(gfxCtx);
+
+    POLY_OPA_DISP = game_play_set_fog(play, POLY_OPA_DISP);
+    POLY_XLU_DISP = game_play_set_fog(play, POLY_XLU_DISP);
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void setupViewer(PlayState* play) {
+    showView(&play->unk_1938, 0xF, play);
+}
+
+void setupViewMatrix(PlayState* play, GraphicsContext* __gfxCtx, GraphicsContext* gfxCtx2) {
```

Comment:
Could you add a TODO for the `__gfxCtx`? so we don't forget to revisit this in the future